### PR TITLE
Fyle 85zrkgg35 Code Coverage :Handle duplicates service

### PR DESCRIPTION
**Pull request by CODEX108**

### CODE REVIEW

File: src/app/core/mock-data/handle-duplicates.service.data.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    transaction_ids: ['tx2uLjJ0myYh', 'txnjPIZMcwxc'],
+    fields: ['amount', 'currency', 'txn_dt'],
+  },
+  {
+    transaction_ids: ['tx5CixxvxeAF', 'txCyV7WCXXqD'],
+    fields: ['amount', 'currency', 'txn_dt'],
+  },
+];
+
+export const handleDuplicatesDataResponse = [
```
Comment (Dimple16) [inline src/app/core/mock-data/handle-duplicates.service.data.ts:22]: type of `handleDuplicatesDataResponse`?

Comment (Dimple16) [reply src/app/core/mock-data/handle-duplicates.service.data.ts:22]: why use 2 variables for the same data? you ca use `handleDuplicatesData` itself?

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,70 @@
...
   it('should be created', () => {
-    expect(service).toBeTruthy();
+    expect(handleDuplicatesService).toBeTruthy();
+  });
+
+  it('getDuplicateSets() : should get all the duplicate sets', (done) => {
+    apiService.get.and.returnValue(of(handleDuplicatesDataResponse));
+    handleDuplicatesService.getDuplicateSets().subscribe((res) => {
+      expect(handleDuplicatesData).toEqual(res);
+      expect(apiService.get).toHaveBeenCalledWith('/transactions/duplicates/sets');
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:34]: `expect(apiService.get).toHaveBeenCalledOnceWith('/transactions/duplicates/sets');`

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,70 @@
...
+      done();
+    });
+  });
+
+  it('getDuplicatesByExpense() : should call the API  with the correct parameters and return the expected data from the API', (done) => {
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:39]: `should get the duplicates by expense`

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 4]
```diff
@@ -1,16 +1,70 @@
...
+    handleDuplicatesService.getDuplicatesByExpense(txnId);
+
+    expect(apiService.get).toHaveBeenCalledWith('/transactions/duplicates/sets', {
+      params: {
+        txn_id: txnId,
+      },
+    });
+    expect(apiService.get).toHaveBeenCalledTimes(1);
+
+    apiService.get.and.returnValue(of(handleDuplicatesDataResponse));
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:50]: why is this required again?

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 5]
```diff
@@ -1,16 +1,70 @@
...
+    handleDuplicatesService.getDuplicateSets().subscribe((res) => {
+      expect(handleDuplicatesData).toEqual(res);
+      expect(apiService.get).toHaveBeenCalledWith('/transactions/duplicates/sets');
+      done();
+    });
+  });
+
+  it('getDuplicatesByExpense() : should call the API  with the correct parameters and return the expected data from the API', (done) => {
+    const txnId = 'txaiCW1efU0n';
+    handleDuplicatesService.getDuplicatesByExpense(txnId);
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:41]:
```
handleDuplicatesService.getDuplicatesByExpense(txnId).subscribe((res) => {
   expect(res).toEqual(handleDuplicatesDataResponse);
   expect(apiService.get).toHaveBeenCalledOnceWith('/transactions/duplicates/sets', {
      params: {
        txn_id: txnId,
      },
    });
    done();
 });
```

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 6]
```diff
@@ -1,16 +1,70 @@
...
+
+    apiService.get.and.returnValue(of(handleDuplicatesDataResponse));
+    handleDuplicatesService.getDuplicatesByExpense(txnId).subscribe((res) => {
+      expect(res).toEqual(handleDuplicatesData);
+    });
+    expect(apiService.get).toHaveBeenCalledTimes(2);
+    done();
+  });
+
+  it('dismissAll() : should call the API service with correct params and dismiss all the duplicates', (done) => {
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:58]: `should dismiss all duplicates`

File: src/app/core/services/handle-duplicates.service.spec.ts [hunk 7]
```diff
@@ -1,16 +1,70 @@
...
+    const duplicateSetTransactionIds = ['tx5CixxvxeAF', 'txCyV7WCXXqD'];
+    const transactionIds = ['tx5CixxvxeAF', 'txCyV7WCXXqD'];
+    handleDuplicatesService.dismissAll(duplicateSetTransactionIds, transactionIds);
```
Comment (Dimple16) [inline src/app/core/services/handle-duplicates.service.spec.ts:61]:
```
apiService.post.and.returnValue(of(null));
const duplicateSetTransactionIds = ['tx5CixxvxeAF', 'txCyV7WCXXqD'];
const transactionIds = ['tx5CixxvxeAF', 'txCyV7WCXXqD'];

handleDuplicatesService.dismissAll(duplicateSetTransactionIds, transactionIds).subscribe((res) => {
  expect(res).toBeNull();
  expect(apiService.post).toHaveBeenCalledOnceWith('/transactions/duplicates/dismiss', {
      duplicate_set_transaction_ids: duplicateSetTransactionIds,
      transaction_ids: transactionIds,
    });
    done();
});
```

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/services/handle-duplicates.service.spec.ts (hunk 4 above)
Comment (CODEX108) [reply src/app/core/services/handle-duplicates.service.spec.ts]: No it isn't required again I mentioned it twice by mistake

**Review by jayfyle — APPROVED** [review]

File: src/app/core/services/handle-duplicates.service.spec.ts (hunk 3 above)
Comment (Dimple16) [reply src/app/core/services/handle-duplicates.service.spec.ts:39]: this isn't resolved? rephrase the test case description

**Review by Dimple16 — APPROVED** [review]
minor comment, fix and merge
